src/result.rs

```diff
@@ -62,3 +62,12 @@ impl From<io::Error> for IonError {
         }
     }
 }
+
+/// Allows [`std::fmt::Error`]s to be converted to an IonError and propagated using the `?` operator.
+impl From<std::fmt::Error> for IonError {
+    fn from(error: std::fmt::Error) -> Self {
+        IonError::IoError {
+            description: format!("Encountered an IO error: {:?}", error),
```

Comment:
This surfaces `fmt::Error`s (which would happen when encoding scalar values) to the end user as IO issues. This isn't strictly correct, but I don't think it's worth adding another error variant to communicate the distinction. (Our error handling in general could use some more thought, FWIW.)